knowledge-base/custom-localization-dotnet-maui-dataform.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Implementing Custom Localization in .NET MAUI DataForm
+description: Learn how to create and apply custom localization values for the .NET MAUI DataForm component to customize validation error messages.
+type: how-to
+page_title: Customizing Localization for Validation Messages in .NET MAUI DataForm
+slug: custom-localization-dotnet-maui-dataform
+tags: dataform, .net maui, localization, validation, custom error message
+res_type: kb
+ticketid: 1663273
+---
+
+## Environment
+
+<table>
+<tbody>
+<tr>
+<td>Product</td>
+<td>DataForm for .NET MAUI</td>
+</tr>
+<tr>
+<td>Version</td>
+<td>7.0.0</td>
+</tr>
+</tbody>
+</table>
+
+## Description
+
+When working with the [DataForm](https://docs.telerik.com/devtools/maui/controls/dataform/overview) for .NET MAUI, you might need to customize the error messages displayed for validation errors. Specifically, you want to change the default range validation error message to a custom one. 
+
+This KB article also answers the following questions:
+- How to customize error messages in .NET MAUI DataForm?
+- How to use custom localization with DataForm in .NET MAUI?
+- How to change the validation error messages in DataForm?
+
+## Solution
+
+To customize the error messages in the DataForm component, create a custom localization manager by extending the `TelerikLocalizationManager` class. Override the `GetString` method to return custom messages for specific keys. 
+
+Here is how to implement a custom localization manager:
+
+```csharp
+public class CustomTelerikLocalizationManager : TelerikLocalizationManager
+{
+    public override string GetString(string key)
+    {
+        if (key == "DataForm_RangeValidationError")
+        {
+            return "My error message";
+        }
+
+        return base.GetString(key);
+    }
+}
+```
+
+After implementing the custom localization manager, assign it to the `TelerikLocalizationManager.Manager` property before initializing the components:
+
+```csharp
+TelerikLocalizationManager.Manager = new CustomTelerikLocalizationManager();
+InitializeComponent();
+```
+
+### Defining Validation Rules
```

Comment:
"Defining Validation Rules" is a related topic but placing it in the same KB as the main task (changing the validation messages) isn't a good practice. Each KB must have one one topic. Furthermore, this kind of information belongs in the "Validation" article of the component as it doesn't seem to be something custom that requires a KB.